### PR TITLE
RHCLOUD-33802 Migrate clean job to a postgres container

### PR DESCRIPTION
**Pull request by josejulio**
## What?

RHCLOUD-33802 Start using a postgres image to run the clean jobs

## Why?
Latest base images prevent us from installing postgres12 and we had to find other way to do it.

## How?
Using volumes to mount our scripts in the postgresql image instead of building a dedicated image

## Testing
No

## Anything Else?
- Checking what else could be removed because of this change
- I'll add comments linking to names that were obtained from other repositories/sources.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: deployments/clowdapp.yml [hunk 1]
```diff
@@ -120,10 +120,34 @@ objects:
...
       podSpec:
-        image: quay.io/cloudservices/payload-tracker-go:${IMAGE_TAG}
+        image: quay.io/cloudservices/postgresql-rds:12
         restartPolicy: Never
-        command:
-          - ./tools/cron-job.sh
+        volumes:
+          - name: playbook-tracker-go-db-cleaner-volume
+            configMap:
+              name: playbook-tracker-go-db-cleaner-config
```
Comment (josejulio) [inline deployments/clowdapp.yml:128]: This `ConfigMap` is defined below

File: deployments/clowdapp.yml [hunk 2]
```diff
@@ -120,10 +120,34 @@ objects:
...
+        volumeMounts:
+          - name: playbook-tracker-go-db-cleaner-volume
+            mountPath: /playbook-tracker-go-db-cleaner
+        command: [ 'sh', '/playbook-tracker-go-db-cleaner/clean.sh' ]
+        env:
+          - name: PGHOST
+            valueFrom:
+              secretKeyRef:
+                name: payload-tracker-db-creds
```
Comment (josejulio) [inline deployments/clowdapp.yml:137]: This secret is taken from [our config](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/insights/payload-tracker/namespaces/payload-tracker-prod.yml#L72).

File: deployments/clowdapp.yml [hunk 3]
```diff
@@ -199,6 +253,18 @@ parameters:
   value: 'true'
 - name: CLEANER_SCHEDULE
   value: "00 10 * * *"
+- name: DB_NAME
+  description: Database name used by the db-cleaner CronJob
+  value: postgres
```
Comment (josejulio) [inline deployments/clowdapp.yml:258]: This one matches our database name in the [database definition](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/terraform/resources/insights/stage/rds/rds-payload-tracker-stage-12.yml#L3).

**Review by tahmidefaz — APPROVED** [review]
The changes looks good to me. If there are no issues, we can merge this in next week. Let it sit in stage for a few days, before promoting to prod. :smile:

**Review by dehort — CHANGES_REQUESTED** [review]
I like this approach.  I think this is a great idea!

I left a couple of comments.  

Thanks!!

File: deployments/clowdapp.yml [hunk 4]
```diff
@@ -120,10 +120,34 @@ objects:
...
+            mountPath: /playbook-tracker-go-db-cleaner
+        command: [ 'sh', '/playbook-tracker-go-db-cleaner/clean.sh' ]
+        env:
+          - name: PGHOST
+            valueFrom:
+              secretKeyRef:
+                name: payload-tracker-db-creds
+                key: ${DB_SECRET_HOSTNAME_KEY}
+          - name: PGDATABASE
+            value: ${DB_NAME}
```
[the remark this answers is not in the document]
Comment (josejulio) [reply deployments/clowdapp.yml:140]: I wasn't aware from that - is there any reference or doc about the database name?

File: deployments/clowdapp.yml [hunk 5]
```diff
@@ -120,10 +120,34 @@ objects:
...
+          - name: PGUSER
+            valueFrom:
+              secretKeyRef:
+                name: virtual-assistant-db
```
[the remark this answers is not in the document]
Comment (josejulio) [reply deployments/clowdapp.yml:144]: yes, copy/pasta mistake, thanks!

[the remark this answers is not in the document]
Comment (josejulio) [reply deployments/clowdapp.yml]: Ref is here: https://gitlab.cee.redhat.com/service/app-interface#manage-rds-databases-via-app-interface-openshiftnamespace-1yml 
Thanks @dehort !

**Comment by josejulio** [comment]
@dehort Just updated the code to address the review. 

Updated to use the `db.name` content from the secret and using `payload-tracker` instead of `playbook-tracker` :sweat_smile:

**Review by dehort — APPROVED** [review]
Looks good to me!

**Comment by dehort** [comment]
Thank you @josejulio !!

**Pull request merged**